src/AppBundle/Controller/UserController.php

```diff
@@ -73,4 +76,27 @@ public function avatarAction($username, Request $request)
 
         return $response;
     }
+
+    /**
+     * @Route("/register/resend-email", name="register_resend_email", methods={"POST"})
+     */
+    public function resendRegistrationEmailAction(Request $request, UserManagerInterface $userManager, MailerInterface $mailer, SessionInterface $session)
+    {
+        $urlCheckEmail = $this->generateUrl('fos_user_registration_check_email');
+        $urlLogin = $this->generateUrl('fos_user_security_login');
+        $emailExists = $request->request->has('email');
+        if ($emailExists === true) {
+            $email = $request->request->get('email');
+            $user = $userManager->findUserByEmail($email);
+            if ($user !== null) {
+                $mailer->sendConfirmationEmailMessage($user);
+                $session->set('fos_user_send_confirmation_email/email', $email);
+                return $this->redirectToRoute('fos_user_registration_check_email');
+            } else {
+                return $this->redirectToRoute('fos_user_security_login');
+            }
+        } else {
```

Comment:
There is no need to put an `else` here, you can put the `return` outside of the `if`